docs/contribute/contribute.adoc

```diff
@@ -4,126 +4,169 @@ This document is meant for new OpenSCAP contributors and it describes how to
 
-NOTE: Since we're fixing an issue number 455 make sure that your commit message
-contain a
+NOTE: Since you're fixing the issue number 455 make sure that your commit
+message contain a
 link:https://help.github.com/articles/closing-issues-via-commit-messages/[keyword]
 which will close the issue automatically when your pull request is merged.
 
-
 Let's say that you've fixed the issue, made a few commits to your local fix_455
 branch and you think it's ready to be review by other contributors. Before you
 push your local changes to your remote forked repository it's necessary to check
-if your changes will be applicable and won't be in a conflict with some work that
-  other contributors could publish while you were working on the fix.
+if your changes will be applicable and won't be in a conflict with some work
+that other contributors could publish while you were working on the fix.
+
+
+== Optional: Write an automated test for your code
+There is a big chance that the code you've fixed or the code that you've added
+has no test coverage. We encourage you to write a new test or extend the
+existing one to cover the changes/additions to OpenSCAP that you have made.
+It is not mandatory, so reviewer will not require you to write a test, but keep
+in mind that providing a test might uncover some unexpected issues with the
+code. We also run these tests on every pull request so adding a test for your
+fix or new feature might prevent someone from breaking it in the future. If you
+decided to add a test please see our
+link:testing.adoc[guide for writing and running tests].
 
 
 == Rebase before pull request
-If some other contributor pushed some code to the maint-1.0 branch while you was
-working of the fix then it's necessary to pull those changes and rebase our fix
-on top of them. First we need to make sure that our local maint-1.0 branch is
-up-to date.
+If some other contributor pushed some code to the `maint-1.2` branch while you
+were working on the fix then it's necessary to fetch those changes and rebase
+your fix on top of them. First you need to make sure that your local `maint-1.2`
+branch is up-to date:
 
 [[app-listing]]
 [source,bash]
 ----
-# add remote name 'upstream' to refer to the upstream's repository
+# add remote name 'upstream' to refer to the upstream's repository (this needs
+# to be done only once - next time you rebase you can skip this step)
 $ git remote add upstream git@github.com:OpenSCAP/openscap.git
-# checkout to branch 'maint-1.0' in your local forked repository
-$ git checkout maint-1.0
-# download and apply any upstream changes to your local maint-1.0 branch
-$ git pull --ff-only upstream maint-1.0
+# checkout to branch 'maint-1.2' in your local forked repository
+$ git checkout maint-1.2
+
+# fetch the branches and their respective commits from the upstream repository
+$ git fetch upstream
+# merge the changes from upstream/maint-1.2 into your local maint-1.2 branch
+$ git merge upstream/maint-1.2
```

Comment:
this is a very verbose way of pulling.

just do: `git checkout maint-1.2; git pull upstream maint-1.2`

Reply:
Ok, I will change it to pull instead of fetch/merge, but basically it does the same thing.